File: src/org/elixir_lang/beam/psi/Module.java
package org.elixir_lang.beam.psi;

import com.intellij.psi.PsiCompiledElement;
import org.elixir_lang.psi.NamedElement;
import org.elixir_lang.psi.call.CanonicallyNamed;

public interface Module extends CanonicallyNamed, NamedElement, PsiCompiledElement {
}

File: src/org/elixir_lang/beam/psi/Module.kt
package org.elixir_lang.beam.psi

interface Module : BeamSymbol {
    /** Never null: the stub always carries one, which is what makes this indexable by name. */
    override fun getName(): String

    fun callDefinitions(): Array<out CallDefinition>
    fun typeDefinitions(): Array<out TypeDefinition>
}
